src/codegen/spirv.zig

```diff
@@ -721,75 +721,19 @@ const DeclGen = struct {
         };
     }
 
-    /// Construct a struct at runtime.
-    /// ty must be a struct type.
-    /// Constituents should be in `indirect` representation (as the elements of a struct should be).
-    /// Result is in `direct` representation.
-    fn constructStruct(self: *DeclGen, ty: Type, types: []const Type, constituents: []const IdRef) !IdRef {
-        assert(types.len == constituents.len);
-        // The Khronos LLVM-SPIRV translator crashes because it cannot construct structs which'
-        // operands are not constant.
-        // See https://github.com/KhronosGroup/SPIRV-LLVM-Translator/issues/1349
-        // For now, just initialize the struct by setting the fields manually...
-        // TODO: Make this OpCompositeConstruct when we can
-        const ptr_composite_id = try self.alloc(ty, .{ .storage_class = .Function });
-        for (constituents, types, 0..) |constitent_id, member_ty, index| {
-            const ptr_member_ty_ref = try self.ptrType(member_ty, .Function);
-            const ptr_id = try self.accessChain(ptr_member_ty_ref, ptr_composite_id, &.{@as(u32, @intCast(index))});
-            try self.func.body.emit(self.spv.gpa, .OpStore, .{
-                .pointer = ptr_id,
-                .object = constitent_id,
-            });
-        }
-        return try self.load(ty, ptr_composite_id, .{});
-    }
-
-    /// Construct a vector at runtime.
-    /// ty must be an vector type.
+    /// Construct a composite value at runtime.
+    /// `ty` must be an struct, vector or array.
     /// Constituents should be in `indirect` representation (as the elements of an vector should be).
     /// Result is in `direct` representation.
```

Comment:
maybe something like
```suggestion
    /// Construct a composite value at runtime. If the parameters are in direct
    /// representation, then the result is also in direct representation. Otherwise,
    /// if the parameters are in indirect representation, then the result is too.
```
? We can change it later too